### ISSUE TITLE
Prevent Editing of Datasets Without Resource Maps

### ISSUE THREAD

**Issue by vchendrix** [NCEAS/metacatui#2588]
Currently, users can edit datasets via the submit endpoint (`submit/<pid>`) even if the dataset has not finished indexing or is in a corrupted state due to a missing resource map. This is problematic as users might unknowingly edit incomplete or corrupted datasets. The metadata editor should enforce a rule that prevents users from editing any dataset that does not have a resource map. 

**Steps to Reproduce:**
1. Save a dataset with a large number of files.
2. Directly access the submit endpoint (`submit/<pid>`) before the indexing is complete.
3. Observe that the user can still edit the dataset.

**Expected Behavior:**
- The metadata editor should not allow users to edit a dataset if it does not have a resource map.

**Actual Behavior:**
- Users can edit the dataset even if it does not have a resource map.

**Proposed Solution:**
- Modify the metadata editor to check for the presence of a resource map before allowing edits. If the resource map is missing, display a message indicating that the dataset cannot be edited until the resource map is available.

**Additional Context:**
- We have a current use case where users are going directly to the `submit` endpoint to edit the datasets when they have not indexed completely causing us to manually recover their resource maps.
- **Enhance Metadata View**:  If the dataset has not been fully indexed, the edit button disappears. It may be that the edit button should not be hidden in this case but maybe grayed out with a message saying that the dataset is not available for editing yet.

**Comment by mbjones** [comment]
Thanks @vchendrix -- while that all makes sense, I think we will also have to balance that with the need to handle datasets without a resource map. Historically, resource maps were not required, and so we have many data packages with just an EML document and linked data files. A design goal for both Morpho and MetacatUI has always been to let users open any of these historic datasets, and have the app bring it forward to the most recent version (in this case, adding a resource map). Another common use is for someone to drop an EML document in Metacat (sans RM), and then want to convert it into a full-fledged package. That said, I do fully agree with you that we need to differentiate the case of "no rm exists" from the case you are describing which is "rm is not yet indexed". Looking forward to the discussion.

**Comment by vchendrix** [comment]
@mbjones Thanks for that explanation.  I am wondering a couple things:

1.  if there is a way to prevent direct access to the `submit/<pid>` endoint and make sure the referrer is the metadata landing page?  This would go a long way to prevent users from editing datasets that have not fully indexed.
2. Also,  on the MetadataView page,  Instead of hiding the the edit button if the resource map is not available, would it be possible to show a disabled edit button with a tooltip or notifiaction on the page that says the dataset is not editable at this time and to check back?

**Comment by mbjones** [comment]
👍 The checking on indexing seems like a good precaution to me. We may have to follow both the RM and EML version chains to discover the PID of the current RM before we can check if it is indexing. e.g., 

```
current EML PID2 --obsoletes--> past EML PID1
EML PID1 --documentedBy--> past RM PID3
past RM PID3 --obsoletedBy--> current RM PID4
check if RM PID4 exists and is being indexed
```

**Comment by robyngit** [comment]
Ok great, then I am going to close this issue as a duplicate of#558 (Check the index queue for the resource map when the editor is opened without one), and make sure that one is prioritized.
